### ISSUE TITLE
💅 ```useNamingConvention``` does not flag UPPER_CASE shorthand object properties

### ISSUE THREAD

**Issue by WaterKat** [biomejs/biome#6371]
### Environment information

```bash
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v22.16.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.2.15"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  true
  GraphQL enabled:              false
  Recommended:                  false
  All:                          false
  Enabled rules:
  style/useNamingConvention
```

### Rule name

lint/style/useNamingConvention

### Playground link

https://biomejs.dev/playground/?indentStyle=space&quoteStyle=single&jsxQuoteStyle=single&lintRules=all&code=YwBvAG4AcwB0ACAAUwBFAFIAVgBJAEMARQBfAE4AQQBNAEUAIAA9ACAAIgBtAHkALQBzAGUAcgB2AGkAYwBlACIAIABhAHMAIABjAG8AbgBzAHQAOwAKAAoAYwBvAG4AcwB0ACAAUwBFAFIAVgBJAEMARQAgAD0AIAB7AAoAIAAgAFMARQBSAFYASQBDAEUAXwBOAEEATQBFACwACgAgACAAUwBFAFIAVgBJAEMARQBfAFAAUgBPAFAARQBSAFQAWQA6ACAAIgBwAHIAbwBwAGUAcgB0AHkAIgAKAH0AIABhAHMAIABjAG8AbgBzAHQAOwAKAAoA

### Expected result

```lint/style/useNamingConvention: 
This 
object property
 name should be in 
camelCase
.```
should be raised for the shorthand property

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
